libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -100,16 +100,7 @@ string YulUtilFunctions::copyToMemoryFunction(bool _fromCalldata)
 		{
 			return Whiskers(R"(
 				function <functionName>(src, dst, length) {
-					let i := 0
-					for { } lt(i, length) { i := add(i, 32) }
-					{
-						mstore(add(dst, i), mload(add(src, i)))
-					}
-					if gt(i, length)
-					{
-						// clear end
-						mstore(add(dst, length), 0)
```

Comment:
I.e. the behavioural change leading to test failures is the removal of this part here. ``copyToMemoryFunction`` is assumed to pad with zeroes and that's not optional... we can either just add this back or maybe it'd also just work to round up the return size of the static call to a multiple of 32 (or add 32 to it), but not entirely sure...

Reply:
Hm, at least I thought this causes actual test failures, not sure I still see them... on the other hand the change needs to be restricted by evm version in any case that might make it clearer to see which tests properly fail...